Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOption.m

```diff
@@ -81,7 +81,20 @@ - (BOOL)hasIcon {
     return self.iconName != nil;
 }
 
-- (BOOL)isEqual:(BPKHorizontalNavigationOption *)other {
+- (BOOL)isEqual:(id)object {
```

Comment:
why do we need to implement this?

Reply:
Because the [implementation](https://github.com/Skyscanner/backpack-ios/blob/4ec2ed34ef0501ebaa98299507c193c499699b10/Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigation.m#L289) relies on it being implemented

Reply:
Is it worth adding a comment to that effect so that ~if~ when someone copies our code for their own implementation, they understand why it's needed?

Reply:
I'll add it to the protocol